js/ui/iconGrid.js

```diff
@@ -759,7 +759,7 @@ const PaginatedIconGrid = new Lang.Class({
         }
 
         let x = box.x1 + leftEmptySpace + this.leftPadding;
-        let y = box.y1 + this.topPadding;
+        let y = box.y1 + this.topPadding + this._spaceBetweenPages / 2;
```

Comment:
It looks like might work when you have multiple pages, but break things when you only have one, as this spacing is not meant to be used for centering things, but to separate multiple pages. For centering things I believe you'd have to implement a `yAlign` property, similar to the `xAlign` one you can see here.